tests/vmi_cloudinit_test.go

```diff
@@ -598,10 +591,16 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 				// Expect that the secret is not present on the vmi itself
 				vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
-				Expect(vmi.Spec.Volumes[idx].CloudInitConfigDrive.UserData).To(BeEmpty())
-				Expect(vmi.Spec.Volumes[idx].CloudInitConfigDrive.UserDataBase64).To(BeEmpty())
-				Expect(vmi.Spec.Volumes[idx].CloudInitConfigDrive.NetworkData).To(BeEmpty())
-				Expect(vmi.Spec.Volumes[idx].CloudInitConfigDrive.NetworkDataBase64).To(BeEmpty())
+
+				for _, volume := range vmi.Spec.Volumes {
+					if volume.CloudInitConfigDrive != nil {
+						Expect(volume.CloudInitConfigDrive.UserData).To(BeEmpty())
```

Comment:
Can you check for presence of the secret ref?

Reply:
I think this is out of slope of this pr. We can do that on a follow up pr, but honestly, I'm not sure it's so important.

Reply:
done